eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClientBuilder.java

```diff
@@ -3,36 +3,75 @@
 
 package com.azure.eventhubs;
 
+import com.azure.core.configuration.BaseConfigurations;
+import com.azure.core.configuration.Configuration;
+import com.azure.core.configuration.ConfigurationManager;
 import com.azure.core.http.policy.RetryPolicy;
 import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
 
+import java.net.InetSocketAddress;
+import java.net.Proxy;
 import java.time.Duration;
 
 /**
  * Builder to create an {@link EventHubClient}.
  */
 public class EventHubClientBuilder {
-    private ConnectionStringBuilder credentials;
+
+    private static final String AZURE_EVENT_HUB_CONNECTION_STRING = "AZURE_EVENT_HUB_CONNECTION_STRING";
+
+    private CredentialInfo credentials;
     private TransportType transport;
     private Duration duration;
     private Scheduler scheduler;
+
     private ProxyConfiguration proxyConfiguration;
     private RetryPolicy retryPolicy;
+    private Configuration configuration;
 
     /**
      * Creates a new instance with the default transport {@link TransportType#AMQP}.
      */
     public EventHubClientBuilder() {
-        transport = TransportType.AMQP;
+        this.transport = TransportType.AMQP;
+    }
+
+    /**
+     * Creates a new {@link EventHubClient} based on the configuration set in this builder.
+     *
+     * @return A new {@link EventHubClient} instance.
+     */
+    public EventHubClient build() {
+        this.configuration = this.configuration == null ? ConfigurationManager.getConfiguration().clone() : this.configuration;
+
+        this.proxyConfiguration = constructDefaultProxyConfiguration(this.configuration);
+
+        if (this.credentials == null) {
+            String connStr = this.configuration.get(AZURE_EVENT_HUB_CONNECTION_STRING);
+            this.credentials = new CredentialInfo(connStr);
+        }
+
+        if (this.duration == null) {
+            this.duration = Duration.ofSeconds(5);
```

Comment:
There is a default in ClientConstants. OperationTimeout. I believe it's 1 minute.